### PR TITLE
chore: use new reloadable config api in router and batchrouter

### PR DESCRIPTION
**Pull request by atzoum**
# Description

- Use new reloadable config api in router and batchrouter
- Drop support for `forceHonorUploadFrequency` since it was never used
- Minor improvements on router tests, when mock calls are missing

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by lvrach — APPROVED** [review]

File: router/batchrouter/handle.go [hunk 1]
```diff
@@ -186,14 +184,14 @@ func (brt *Handle) getWorkerJobs(partition string) (workerJobs []*DestinationJob
 	queryStart := time.Now()
 	queryParams := jobsdb.GetQueryParams{
 		CustomValFilters: []string{brt.destType},
-		JobsLimit:        limit,
-		PayloadSizeLimit: brt.adaptiveLimit(brt.payloadLimit),
+		JobsLimit:        limit.Load(),
```
Comment (Sidddddarth) [inline router/batchrouter/handle.go:187]:
```suggestion
		JobsLimit:        brt.jobQueryBatchSize.Load(),
```

File: router/router_test.go [hunk 2]
```diff
@@ -320,12 +321,14 @@ var _ = Describe("router", func() {
 
 	BeforeEach(func() {
 		conf = config.New()
-		routerUtils.JobRetention = time.Duration(175200) * time.Hour // 20 Years(20*365*24)
+		config.Reset()
+		config.Set("Router.jobRetention", "175200h") // 20 Years(20*365*24)
```
Comment (Sidddddarth) [inline router/router_test.go:325]: shouldn't it be:
```suggestion
		conf.Reset()
		conf.Set("Router.jobRetention", "175200h") // 20 Years(20*365*24)
```

and `conf.Reset()` in `AfterEach`

File: router/router_test.go [hunk 3]
```diff
@@ -686,8 +703,8 @@ var _ = Describe("router", func() {
 			router.Setup(gaDestinationDefinition, logger.NOP, conf, c.mockBackendConfig, c.mockRouterJobsDB, c.mockProcErrorsDB, transientsource.NewEmptyService(), rsources.NewNoOpService(), destinationdebugger.NewNoOpService())
 			router.transformer = mockTransformer
 			router.noOfWorkers = 1
-			router.reloadableConfig.noOfJobsToBatchInAWorker = 5
-			router.reloadableConfig.routerTimeout = time.Duration(0)
+			router.reloadableConfig.noOfJobsToBatchInAWorker = config.GetReloadableIntVar(5, 1, rand.UniqueString(10))
+			router.reloadableConfig.routerTimeout = config.GetReloadableDurationVar(0, time.Nanosecond, rand.UniqueString(10))
```
Comment (Sidddddarth) [inline router/router_test.go:707]:
```suggestion
			conf.Set(...) // before calling setup like in other places?
```

File: router/router_test.go [hunk 4]
```diff
@@ -810,9 +834,9 @@ var _ = Describe("router", func() {
 			router.Setup(gaDestinationDefinition, logger.NOP, conf, c.mockBackendConfig, c.mockRouterJobsDB, c.mockProcErrorsDB, transientsource.NewEmptyService(), rsources.NewNoOpService(), destinationdebugger.NewNoOpService())
 			router.transformer = mockTransformer
 			router.noOfWorkers = 1
-			router.reloadableConfig.noOfJobsToBatchInAWorker = 5
-			router.reloadableConfig.routerTimeout = time.Duration(60) * time.Second
-			router.reloadableConfig.jobIteratorMaxQueries = 1
+			router.reloadableConfig.noOfJobsToBatchInAWorker = config.GetReloadableIntVar(5, 1, rand.UniqueString(10))
+			router.reloadableConfig.routerTimeout = config.GetReloadableDurationVar(60, time.Second, rand.UniqueString(10))
+			router.reloadableConfig.jobIteratorMaxQueries = config.GetReloadableIntVar(1, 1, rand.UniqueString(10))
```
Comment (Sidddddarth) [inline router/router_test.go:839]: same as [above](https://github.com/rudderlabs/rudder-server/pull/3871/files#r1327127125)

File: router/router_test.go [hunk 5]
```diff
@@ -1209,8 +1254,8 @@ var _ = Describe("router", func() {
 			router.Setup(gaDestinationDefinition, logger.NOP, conf, c.mockBackendConfig, c.mockRouterJobsDB, c.mockProcErrorsDB, transientsource.NewEmptyService(), rsources.NewNoOpService(), destinationdebugger.NewNoOpService())
 			router.transformer = mockTransformer
 			router.noOfWorkers = 1
-			router.reloadableConfig.noOfJobsToBatchInAWorker = 5
-			router.reloadableConfig.routerTimeout = time.Duration(math.MaxInt64)
+			router.reloadableConfig.noOfJobsToBatchInAWorker = config.GetReloadableIntVar(5, 1, rand.UniqueString(10))
+			router.reloadableConfig.routerTimeout = config.GetReloadableDurationVar(math.MaxInt64, time.Nanosecond, rand.UniqueString(10))
```
Comment (Sidddddarth) [inline router/router_test.go:1258]: use `conf.Set(...)`?

File: router/router_test.go [hunk 6]
```diff
@@ -903,9 +934,9 @@ var _ = Describe("router", func() {
 			router.transformer = mockTransformer
 
 			router.enableBatching = true
-			router.reloadableConfig.noOfJobsToBatchInAWorker = 3
+			router.reloadableConfig.noOfJobsToBatchInAWorker = config.GetReloadableIntVar(3, 1, rand.UniqueString(10))
 			router.noOfWorkers = 1
-			router.reloadableConfig.routerTimeout = time.Duration(math.MaxInt64)
+			router.reloadableConfig.routerTimeout = config.GetReloadableDurationVar(math.MaxInt64, time.Nanosecond, rand.UniqueString(10))
```
Comment (Sidddddarth) [inline router/router_test.go:939]: use `config.Set(...)`?

**Comment by Sidddddarth** [comment]
LGTM. 
Posted some minor comments

File: router/router_test.go (hunk 2 above)
Comment (atzoum) [reply router/router_test.go:325]: unfortunately this is not possible, needs further refactoring that I am not willing to perform at this stage (global config is used for this specific option)

File: router/router_test.go (hunk 6 above)
Comment (atzoum) [reply router/router_test.go:939]: I tried to perform the change "blindly" as much as possible to make the task easier. I will try to further optimise on the next task, where I will be introducing a `misc.SingleValueLoader`

**Review by Sidddddarth — APPROVED** [review]

**Pull request merged**